packages/core/src/amazonqFeatureDev/session/sessionState.ts

```diff
@@ -266,6 +265,15 @@ abstract class CodeGenBase {
                 case 'predict-failed':
                 case 'debate-failed':
                 case 'Failed': {
+                    // canned errors are not retryable
+                    if (codegenResult.codeGenerationStatusDetail) {
+                        return {
+                            result: 'failed',
+                            message: codegenResult.codeGenerationStatusDetail,
+                            retryable: false,
+                        }
+                    }
+
                     throw new ToolkitError('Code generation failed', { code: 'CodeGenFailed' })
```

Comment:
Won't we also want to know that code generation failed? I think this code will make it so the telemetry failure event never gets reported when we get a canned error

Reply:
This function is invoked from inside a `amazonq_codeGenerationInvoke` telemetry block, so I assume we log the failure if we throw inside the block.

For the canned error case that we don't throw, we set 
```
            telemetry.setCodeGenerationResult(codegenResult.codeGenerationStatus.status)
```
so we will have the information of generations that failed.